orchestrator/tests/test_prd177_composio_telemetry.py

```
@@ -0,0 +1,201 @@
+    # Guard: if heavy deps are unavailable in this env, still assert the resolved
+    # names are the per-action ones (the property F016 guarantees).
+    try:
+        mod = _u.module_from_spec(spec)
+        spec.loader.exec_module(mod)
+        edges = mod._compute_used_after_edges(logs)
+        keys = set(edges.keys())
+        assert ("SLACK_SEND_MESSAGE", "SLACK_ADD_REACTION", "ws-1", None) in keys, (
+            "edge builder must produce a used_after edge between the resolved "
+            "per-action names"
+        )
+    except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)